### PR TITLE
replication documentation

### PR DESCRIPTION
**Pull request by xhoaluu**
Hi @jaimecasero 

Could you help me to review.

### CODE REVIEW

File: docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc [hunk 1]
```diff
@@ -23,6 +23,30 @@ The application servers within a clustered environment will use what is called a
...
+
+Replication strategy can be EarlyDialog or ConfirmDialog. EarlyDialog replication strategy is to replicate data when the dialog goes to EARLY state. 
+In the figure, the EarlyDialog replication supports from step 5 to step 12. When 1xx response (except 100 Trying) with To tag is sent out (step 5), the sip application session, transaction and dialog will be replicated. 
+There are a lot of overhead and the performance shall be decreased in term of EarlyDialog replication stratergy. 
+
+ConfirmDialog replication strategy is to replicate data when the dialog goes to CONFIRM state (step 7 to step 12). When 200 OK is sent out (step 7), the sip application session, dialog and sip session will be replicated.
+
+[.text-center]
+.Figure 1
+image::images/replication-seq-diagram.png[align="center"]
```
Comment (jaimecasero) [inline docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc]: wed better insert diagram, not resulting image. Check opensource playbbok https://docs.google.com/document/d/1RZz2nd2ivCK_rg1vKX9ansgNF6NpK_PZl81GxZ2MSnM/edit#heading=h.odaumotkzac9 section inserting UML diagramas.

The idea is to upload the diagram plantuml source into a file in the repo, and the use the resulting url to embed the diagram. There final picture will be automatically generated. “http://www.plantuml.com/plantuml/proxy?src=<urlToYourPlantUmlFile>”

File: docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+@startuml
```
Comment (jaimecasero) [inline docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml]: this is what i meant, this files needs to be saved as it is. THe previous png file shouldnt be part of the repository.

File: docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+UA1 -> SipServlet: INVITE             
+hnote over SipServlet: 1
+UA1 <- SipServlet: Trying
+hnote over SipServlet: 2
+UA2 <- SipServlet: INVITE
+hnote over SipServlet: 3
+UA2 -> SipServlet: Ringing
+hnote over SipServlet: 4
+UA1 <- SipServlet: Ringing
+hnote over SipServlet: 5
```
Comment (jaimecasero) [inline docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml]: Please add note on earlyDialog replciation as explained in http://plantuml.com/sequence-diagram
note left
	On EARLY_DIALOG mode, dialog/sipSession will be replicated
end note

File: docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+UA2 -> SipServlet: OK
+hnote over SipServlet: 6
+UA1 <- SipServlet: OK
+hnote over SipServlet: 7
```
Comment (jaimecasero) [inline docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml]: add same kind of note for confirmedDailog note

File: docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc [hunk 5]
```diff
@@ -23,6 +23,30 @@ The application servers within a clustered environment will use what is called a
...
+However, when a synchronous replication returns successfully, the caller knows for sure that all modifications have been applied to all cache instances, whereas this is not be the case with asynchronous replication. 
+With asynchronous replication, errors are simply written to a log. Even when using transactions, a transaction may succeed but replication may not succeed on all cache instances.
+
+Replication strategy can be EarlyDialog or ConfirmDialog. EarlyDialog replication strategy is to replicate data when the dialog goes to EARLY state. 
+In the figure, the EarlyDialog replication supports from step 5 to step 12. When 1xx response (except 100 Trying) with To tag is sent out (step 5), the sip application session, transaction and dialog will be replicated. 
+There are a lot of overhead and the performance shall be decreased in term of EarlyDialog replication stratergy. 
+
+ConfirmDialog replication strategy is to replicate data when the dialog goes to CONFIRM state (step 7 to step 12). When 200 OK is sent out (step 7), the sip application session, dialog and sip session will be replicated.
```
Comment (jaimecasero) [inline docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc]: is sipappsession replciated previously on CONFIRMEDDIALOG mode?

**Comment by deruelle** [comment]
@xhoaluu were you able to move forward on this ?

**Comment by xhoaluu** [comment]
Hi Jean,

Currently, I’m busy with support activities. I will come back to update it as soon as I can.

Thanks and Best Regards
Hoan HL

File: docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc (hunk 5 above)
Comment (xhoaluu) [inline docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc]: Confirmed, on CONFIRMEDDIALOG mode, the sip app session is replicated after 100 trying is sending out to the caller. Updated the state.

File: docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc (hunk 1 above)
Comment (jaimecasero) [inline docs/sources-asciidoc/src/main/asciidoc/concept-section-SS_for_JBoss-Clustering_Support.adoc]: @xhoaluu  when you invoke maven on this module, you can check the generated image on target directory. Make sure the image looks good, and outcome is as expected

File: docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml (hunk 4 above)
Comment (xhoaluu) [inline docs/sources-asciidoc/src/main/asciidoc/uml/replication-seq-diagram.plantuml]: added

**Comment by jaimecasero** [comment]
As discussed im merging this contribution with further changes we are introducing to the doc. Thanks @xhoaluu  for the diagrams..

**Pull request closed without merging**